src/Integration.Vsix.UnitTests/Settings/SonarLintSettingsTests.cs

```diff
@@ -22,354 +22,287 @@
-            this.settingsStore.AssertCollectionPropertyCount(collection, 0);
-
-            // Test case 1: new property
-            // Act
-            testSubject.SetValue(propertyKey, true);
+            var actual = testSubject.GetValueOrDefault("key1", expected);
+            actual.Should().Be(expected);
+        }
 
-            // Assert
-            this.settingsStore.AssertCollectionPropertyCount(collection, 1);
-            this.settingsStore.AssertBoolean(collection, propertyKey, true);
+        [TestMethod]
+        [DataRow("a key", true)]
+        [DataRow("another key", false)]
+        public void SetValue_Bool(string propertyName, bool valueToSet)
+        {
+            var settingsStore = new Mock<WritableSettingsStore>();
 
-            // Test case 2: overwrite existing property
-            // Act
-            testSubject.SetValue(propertyKey, false);
+            var testSubject = CreateTestSubject(settingsStore.Object);
 
-            // Assert
-            this.settingsStore.AssertCollectionPropertyCount(collection, 1);
-            this.settingsStore.AssertBoolean(collection, propertyKey, false);
+            testSubject.SetValue(propertyName, valueToSet);
+            
+            CheckPropertySet(settingsStore, propertyName, valueToSet);
         }
 
         [TestMethod]
-        public void IntegrationSettings_SetValue_Bool_NoStore()
+        public void SetValue_NoStore_Bool()
         {
-            // Arrange
-            SonarLintSettings testSubject;
-            using (new AssertIgnoreScope())
-            {
-                testSubject = this.CreateTestSubject(storeLoadFailure: true);
-            }
+            var testSubject = CreateTestSubject(storeToReturn: null);
 
             // Act + Assert (no store -> no exception)
             Action act = () => testSubject.SetValue("key1", false);
 
             act.Should().NotThrow();
         }
 
-        [TestMethod]
-        public void IntegrationSettings_GetValueOrDefault_String()
-        {
-            // Arrange
-            SonarLintSettings testSubject = this.CreateTestSubject();
+        #endregion
 
-            // Test case 1: exists -> value
-            // Arrange
-            string expected1 = "value";
-            this.settingsStore.SetString(SonarLintSettings.SettingsRoot, "key1", expected1);
+        #region String method tests
 
-            // Act
-            string actual1 = testSubject.GetValueOrDefault("key1", "default");
-
-            // Assert
-            actual1.Should().Be(expected1, "Did not load existing value");
+        [TestMethod]
+        [DataRow("val1", "unused default")]
+        [DataRow("xxx", "xxx")]
+        public void GetValueOrDefault_String(string valueToReturn, string defaultValueSuppliedByCaller)
+        {
+            var store = new Mock<WritableSettingsStore>();
+            store.Setup(x => x.GetString(SonarLintSettings.SettingsRoot, "aProp", defaultValueSuppliedByCaller)).Returns(valueToReturn);
```

Comment:
This is effectively the only real change in the test from the boolean/int version: we're mocking a different method.